### ISSUE TITLE
Data fetch markers being left after union merge with no data

### ISSUE THREAD

**Issue by gardnervickers** [untangled-web/client#54]
https://gist.github.com/gardnervickers/687da840630c07d1e4f9721201a051c3
Here's the devcard app state
<img width="975" alt="devcardstate" src="https://cloud.githubusercontent.com/assets/2684011/21202634/288bb038-c21d-11e6-99bf-97c9aaac526a.png">

The gist contains a repro, tested on `0.6.0` and `0.6.0-SNAPSHOT`. Also happens with the cookbook tabbed union example.

**Comment by awkay** [comment]
OK, changing this line:

https://github.com/untangled-web/untangled-client/blob/feature/issue-54/src/untangled/client/impl/om_plumbing.cljs#L198

fixes the union issue, but breaks some of the ident join stuff. I cannot remember exactly why we marked things as we did, so I'll want to talk it over with @Pancia to see how we go about fixing it. We're definitely missing some test cases around unions on this particular example.
